File: 347-top-k-frequent-elements/top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        vector<pair<int,int>> ans;
        unordered_map<int,int> mp;
        for(int i : nums){
            mp[i]++;
        }
        for(int i = 0;i < nums.size();i++){
            ans.push_back({mp[nums[i]],nums[i]});
            mp[nums[i]] = 0;
        }
        sort(ans.begin(),ans.end());
        reverse(ans.begin(),ans.end());
        vector<int> result;
        for(int i = 0;i < k;i++){
            result.push_back(ans[i].second);
        }
        return result;
    }
};

File: 347-top-k-frequent-elements/top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer,Integer> mp = new HashMap<>();
        for(int i : nums){
            mp.put(i,mp.getOrDefault(i,0)+1);
        }
        ArrayList<int[]> ans = new ArrayList<>();
        for(int i = 0;i < nums.length;i++){
            ans.add(new int[]{mp.get(nums[i]),nums[i]});
            mp.put(nums[i],0);
        }

        ans.sort((a,b) -> b[0]-a[0]);
        int[] res = new int[k];
        for(int i = 0;i < k;i++){
            res[i] = ans.get(i)[1];
        }
        return res;
    }
}